src/Sarif.Viewer.VisualStudio/Sarif/Run.extensions.cs

```diff
@@ -74,9 +74,13 @@ public static bool TryGetRule(this Run run, string ruleId, string ruleKey, out I
                         helpUri = new Uri(RuleMetadata[ruleId]["url"].Value<string>());
                     }
 
+                    // PREfast rules don't have or need a "help" property; they all have online documentation.
+                    // The rule.help property is only useful in scenarios where there's no online documentation.
+                    string help = null;
+
                     if (ruleName != null || helpUri != null)
                     {
-                        rule = new Rule(ruleId, ruleName, null, null, null, RuleConfiguration.Unknown, ResultLevel.Warning, helpUri, null);
```

Comment:
>null [](start = 58, length = 4)

do me a favor, at the first null literal, use a named arg instead (which will require you to name all remaining args).

make this a multiline statement. After these two things, you won't need the meaningless local initialized to null in order to make this code readable.